docs/user-guide/migrating-to-8.0.0.md

```diff
@@ -0,0 +1,243 @@
+        "functions": "never"
+    }]
+}
+```
+
+**Related issue(s):** [#13739](https://github.com/eslint/eslint/issues/13739)
+
+## <a name="directives"></a> Unused disable directives are now fixable
+
+In ESLint v7.0.0, using both `--report-unused-disable-directives` and `--fix` on the command line would fix only rules but leave unused disable directives in place. In ESLint v8.0.0, this combination of command-line options will result in the unused disable directives being removed.
+
+**To address:** If you are using `--report-unused-disable-directives` and `--fix` together on the command line, and you don't want unused disable directives to be removed, add `--fix-type problem,suggestion,layout` as a command line option.
+
+**Related issue(s):** [#11815](https://github.com/eslint/eslint/issues/11815)
+
+## <a name="eslint-recommended"></a> `eslint:recommended` has been updated
+
+Three new rules have been enabled in the `eslint:recommended` preset.
+
+- [`no-loss-of-precision`](https://eslint.org/docs/rules/no-loss-of-precision)
+- [`no-nonoctal-decimal-escape`](https://eslint.org/docs/rules/no-nonoctal-decimal-escape)
+- [`no-unsafe-optional-chaining`](https://eslint.org/docs/rules/no-unsafe-optional-chaining)
+- [`no-useless-backreference`](https://eslint.org/docs/rules/no-useless-backreference)
+
+**To address:** Fix errors or disable these rules.
+
+**Related issue(s):** [#14673](https://github.com/eslint/eslint/issues/14673)
+
+
+## <a name="suggestions"></a> Rules require `meta.hasSuggestions` to provide suggestions
+
+In ESLint v7.0.0, rules that [provided suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions) did not need to let ESLint know. In v8.0.0, rules providing suggestions need to set their `meta.hasSuggestions` to `true`. This informs ESLint that the rule intends to provide suggestions. Without this property, any attempt to provide a suggestion will result in an error.
+
+**To address:** If your rule provides suggestions, add `meta.hasSuggestions` to the object, such as:
+
+```js
+module.exports = {
+    meta: {
+        hasSuggestions: true
+    },
+    create(context) {
+        // your rule
+    }
+};
+```
+
+**Related issue(s):** [#14312](https://github.com/eslint/eslint/issues/14312)
+
+## <a name="fixes"></a> Rules require `meta.fixable` to provide fixes
+
+In ESLint v7.0.0, rules that were written as a function (rather than object) were able to provide fixes. In ESLint v8.0.0, only rules written as an object are allowed to provide fixes and must have a `meta.fixable` property set to either `"code"` or `"whitespace"`.
+
+**To address:** If your rule makes fixes and is written as a function, such as:
+
+```js
+module.exports = function(context) {
+    // your rule
+};
+```
+
+Then rewrite your rule in this format:
+
+```js
+module.exports = function(context) {
+    meta: {
+        fixable: "code" // or "whitespace"
+    },
+    create(context) {
+        // your rule
+    }
+};
+```
+
```

Comment:
```suggestion

The [eslint-plugin/require-meta-fixable](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/require-meta-fixable.md) rule can automatically fix and enforce that your rules are properly specifying `meta.fixable`.

The [eslint-plugin/prefer-object-rule](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/prefer-object-rule.md) rule can automatically fix and enforce that your rules are written with the object format instead of the deprecated function format.


```

Reply:
I’d like the rest of the TSC to weigh in here. We haven’t traditionally been referring people to outside plugins, so let’s be sure everyone is on board with this approach.

Reply:
Sounds good to hear from others too. If @not-an-aardvark / @aladdin-add agreed, I actually think it would make sense for [eslint-plugin-eslint-plugin](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin) to join the ESLint organization on Github and receive first-party support, given that it's used heavily by ESLint core as well as critical to the health/progress of the entire ESLint plugin ecosystem.

Reply:
FWIW I'd be fine with that if @aladdin-add and the ESLint team are on board with it.

Reply:
same as @not-an-aardvark. 😄

Reply:
@bmish we don't generally adopt already-existing projects, and I can't see us doing so in this case. We can barely keep up with all of the projects we are already maintaining. We've been actively deprecating projects and handing them off to try to relieve the burden (deprecated Doctrine, transitioned typescript-eslint to a separate team, and looking to EOL eslint-release). We'd much rather support an ecosystem where we don't have to maintain all of the useful plugins.

Reply:
Fair enough, thanks for the history lesson! I still do think it would be valuable to reference it more in our docs (pending on what the TSC thinks).